includes/blocks/site-automation-block/register.php

```diff
@@ -56,9 +54,27 @@ function render_block_callback( $attributes, $content, $block ) {
 	 */
 	$bypass_cache = apply_filters( 'sophi_bypass_curated_posts_cache', false, $page_name, $widget_name );
 
-	$curated_posts = get_transient( $curated_posts_transient_key );
+	$sophi_cached_response = new \WP_Query(
```

Comment:
In this case I think we should still do the query. The most important check is if `$bypass_cache === true`, and the next one should be if there's `$curated_posts` does it makes sense @cadic  ?

Reply:
@oscarssanchez thank you for explaining. I mean we could make the query optional.

1. When `$bypass_cache === true`, we will execute contents of if statement in 78:90 regardless the `$curated_posts` initial value, it will be overwrited with `get_posts`.
2. When `$bypass_cache === false`, this if statement will only be executed if `$curate_posts` is negative.

This means to me that we only need to take care about the initial value for `$curated_posts` when `$bypass_cache === false`

```php
$curated_posts = false;

if ( false === $bypass_cache ) {
	$sophi_cached_response = new \WP_Query(
		[
			'post_name__in'          => [ "sophi-site-automation-data-{$page_name}-{$widget_name}" ],
			'post_type'              => 'sophi-response',
			'posts_per_page'         => 1,
			'fields'                 => 'ids',
			'no_found_rows'          => true,
			'update_post_term_cache' => false
		]
	);

	if ( $sophi_cached_response->found_posts ) {
		$last_update = get_post_meta( $sophi_cached_response->posts[0], 'sophi_site_automation_last_updated', true );

		if ( $last_update + 5 * MINUTE_IN_SECONDS > time() ) {
			$curated_posts = get_post_meta( $sophi_cached_response->posts[0], 'sophi_site_automation_data' );
		}
	}
}
```

Reply:
@oscarssanchez was ^ this addressed in your latest commits?

Reply:
@cadic Thanks for clarifying, that makes sense to me! I have addressed this here: https://github.com/globeandmail/sophi-for-wordpress/pull/298/commits/c40cf6e63bb7b3bb4aa28a3f54b3ea1d036a39e9